File: .tmp_unused_imports.js
const fs = require('fs');
const path = require('path');
const file = path.join(process.cwd(), 'src/components/project-data.ts');
const txt = fs.readFileSync(file, 'utf8');
const importLines = txt.split(/\r?\n/).filter(l => l.trim().startsWith('import '));
const names = [];
for (const l of importLines) {
  const m = l.match(/import\s+([^\s,{]+)\s+from\s+['"][^'"]+['"]/)
    || l.match(/import\s+\{\s*([^\}]+)\s*\}\s+from\s+['"][^'"]+['"]/);
  if (!m) continue;
  m[1].split(',').map(s => s.trim()).filter(Boolean).forEach(n => names.push(n));
}
const uniqueNames = [...new Set(names)];
const unused = [];
for (const name of uniqueNames) {
  const regex = new RegExp('\\b' + name + '\\b', 'g');
  const count = (txt.match(regex) || []).length;
  if (count <= 1) unused.push({name, count});
}
console.log(JSON.stringify(unused, null, 2));

File: .tmp_unused_imports.py
import pathlib
import re

path = pathlib.Path('src/components/project-data.ts')
text = path.read_text(encoding='utf-8')
imports = []
for line in text.splitlines():
    line = line.strip()
    if not line.startswith('import '):
        continue
    m = re.match(r'import\s+([^\s,{]+)\s+from\s+["\"][^"\"]+["\"]', line)
    if not m:
        m = re.match(r'import\s+\{\s*([^\}]+)\s*\}\s+from\s+["\"][^"\"]+["\"]', line)
    if not m:
        continue
    names = [n.strip() for n in m.group(1).split(',') if n.strip()]
    for name in names:
        imports.append(name)

unique = []
for name in imports:
    if name not in unique:
        unique.append(name)
unused = []
for name in unique:
    count = len(re.findall(r'\b' + re.escape(name) + r'\b', text))
    if count <= 1:
        unused.append({'name': name, 'count': count})
print(unused)